### DISCUSSION TITLE
[Request] XFDesktop Integration

### DISCUSSION THREAD

**Discussion by moriel5** [BuddiesOfBudgie/budgie-desktop#645] [Ideas]
Budgie version
>Budgie 10.9.2

Use-cases
>Useful for those who prefer a more traditional desktop experience, similar to when Nautilus used to provide the file-management duties.

Proposal
>Add support for XFDesktop (requires Thunar to be installed as well, which in my subjective opinion, does work much better with Budgie than Nemo or Nautilus), similarly to how DesktopFolders is supported (though that specific integration, at list in my experience, was always broken).

References
>None

**Comment by fossfreedom** [comment]
Perhaps expand your thoughts what you mean by integrate? How?

**Comment by moriel5** [comment]
Currently, there are some quirks, such as context menus being shorter on the desktop, XFDesktop competing with Budgie Desktop for rendering the wallpaper and desktop icons (it's a waste to simultaneously render the same thing twice) and the need to manually set XFDesktop to launch at boot (which I believe adds some delay, vs. just have it start together with Budgie Desktop).

I'm not sure how much of this can be alleviated by proper integration (which could be set via DConf, like with DesktopFolders)(at least the wallpaper thing may require upstream changes), but this could at least help to some degree, and also make it easier for distribution maintainers who wish to provide an experience that utilizes Budgie, XFDesktop and Thunar as defaults.

**Comment by fossfreedom** [comment]
I have converted this into a discussion item.

It does sound like there is a need to work with the xfdesktop maintainers to understand where there are potential conflicts that you have identified and more importantly how complex to make those changes. Equally important is to understand if the xfdesktop maintainers would be willing to entertain such changes by making one or more PRs.
